lightning/src/ln/router.rs

```diff
@@ -1864,4 +1870,771 @@ mod tests {
+			short_channel_id,
+			is_permanent: false
+		};
+
+		router.handle_htlc_fail_channel_update(&channel_close_msg);
+
+		{
+			// Non-permanent closing just disables a channel
+			let network = router.network_map.write().unwrap();
+			match network.channels.get(&channel_key) {
+				None => panic!(),
+				Some(channel_info) => {
+					assert!(!channel_info.one_to_two.enabled);
+					assert!(!channel_info.two_to_one.enabled);
+				}
+			}
+		}
+
+		let channel_close_msg = HTLCFailChannelUpdate::ChannelClosed {
+			short_channel_id,
+			is_permanent: true
+		};
+
+		router.handle_htlc_fail_channel_update(&channel_close_msg);
+
+		{
+			// Permanent closing deletes a channel
+			let network = router.network_map.write().unwrap();
+			assert_eq!(network.channels.len(), 0);
+			// Nodes are also deleted because there are no associated channels anymore
+			assert_eq!(network.nodes.len(), 1);
+		}
+
+		// TODO: Test HTLCFailChannelUpdate::NodeFailure, which is not implemented yet.
+	}
+
+	#[test]
+	fn getting_next_channel_announcements() {
+		let (secp_ctx, _, router) = create_router();
+		let node_1_privkey = &SecretKey::from_slice(&[42; 32]).unwrap();
+		let node_2_privkey = &SecretKey::from_slice(&[41; 32]).unwrap();
+		let node_id_1 = PublicKey::from_secret_key(&secp_ctx, node_1_privkey);
+		let node_id_2 = PublicKey::from_secret_key(&secp_ctx, node_2_privkey);
+		let node_1_btckey = &SecretKey::from_slice(&[40; 32]).unwrap();
+		let node_2_btckey = &SecretKey::from_slice(&[39; 32]).unwrap();
+
+		let short_channel_id = 1;
+		let chain_hash = genesis_block(Network::Testnet).header.bitcoin_hash();
+		let channel_key = NetworkMap::get_key(short_channel_id, chain_hash);
+
+		// No channels yet.
+		let next_announcements = router.get_next_channel_announcements(channel_key, 10);
+		assert_eq!(next_announcements.len(), 0);
+
+		{
+			// Announce a channel we will update
+			let unsigned_announcement = UnsignedChannelAnnouncement {
+				features: ChannelFeatures::empty(),
+				chain_hash,
+				short_channel_id,
+				node_id_1,
+				node_id_2,
+				bitcoin_key_1: PublicKey::from_secret_key(&secp_ctx, node_1_btckey),
+				bitcoin_key_2: PublicKey::from_secret_key(&secp_ctx, node_2_btckey),
+				excess_data: Vec::new(),
+			};
+
+			let msghash = hash_to_message!(&Sha256dHash::hash(&unsigned_announcement.encode()[..])[..]);
+			let valid_channel_announcement = ChannelAnnouncement {
+				node_signature_1: secp_ctx.sign(&msghash, node_1_privkey),
+				node_signature_2: secp_ctx.sign(&msghash, node_2_privkey),
+				bitcoin_signature_1: secp_ctx.sign(&msghash, node_1_btckey),
+				bitcoin_signature_2: secp_ctx.sign(&msghash, node_2_btckey),
+				contents: unsigned_announcement.clone(),
+			};
+			match router.handle_channel_announcement(&valid_channel_announcement) {
+				Ok(_) => (),
+				Err(_) => panic!()
+			};
+		}
+
+		// No updates.
+		let next_announcements = router.get_next_channel_announcements(channel_key, 2);
```

Comment:
Oops. I actually think this is a bug, and have a commit to fix it that I haven't gotten around to PR'ing yet.

Reply:
Alright, so should we wait when you merge that first then?

Reply:
Up to you. You're welcome to take the top commit from https://github.com/TheBlueMatt/rust-lightning/tree/2020-04-relay-no-node in this PR or I can do it separately and you can drop the relevant commit from this PR.